app/controllers/image_search_controller.rb

```diff
@@ -2,34 +2,51 @@ class ImageSearchController < ::ApplicationController
   before_filter :find_resource
 
   def auto_complete_repository_name
-    render :text => (use_hub? ? hub_image_exists?(params[:search]) :
-        registry_image_exists?(params[:search])).to_s
+    catch_network_errors do
+      render :text => (use_hub? ? hub_image_exists?(params[:search]) :
+          registry_image_exists?(params[:search])).to_s
+    end
   end
 
   def auto_complete_image_tag
-    # This is the format jQuery UI autocomplete expects
-    tags = use_hub? ? hub_auto_complete_image_tags(params[:search]) :
-        registry_auto_complete_image_tags(params[:search])
-    respond_to do |format|
-      format.js do
-        tags.map! { |tag| { :label => CGI.escapeHTML(tag), :value => CGI.escapeHTML(tag) } }
-        render :json => tags
+    catch_network_errors do
+      # This is the format jQuery UI autocomplete expects
+      tags = use_hub? ? hub_auto_complete_image_tags(params[:search]) :
+          registry_auto_complete_image_tags(params[:search])
+      respond_to do |format|
+        format.js do
+          tags.map! { |tag| { :label => CGI.escapeHTML(tag), :value => CGI.escapeHTML(tag) } }
+          render :json => tags
+        end
       end
     end
   end
 
   def search_repository
-    repositories = use_hub? ? hub_search_image(params[:search]) :
-                              registry_search_image(params[:search])
-    respond_to do |format|
-      format.js do
-        render :partial => 'repository_search_results',
-               :locals  => { :repositories => repositories,
-                             :use_hub => use_hub? }
+    catch_network_errors do
+      repositories = use_hub? ? hub_search_image(params[:search]) :
+                                registry_search_image(params[:search])
+      respond_to do |format|
+        format.js do
+          render :partial => 'repository_search_results',
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.